pkg/service/utgen/gen.go

```diff
@@ -69,23 +72,58 @@ func NewUnitTestGenerator(srcPath, testPath, reportPath, cmd, dir, coverageForma
 }
 
 func (g *UnitTestGenerator) Start(ctx context.Context) error {
+	// Signal handling for Ctrl+C (SIGINT)
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
```

Comment:
rather than redefining this why don't we use the pre-defined context?


Like in main.go we've a sigchan handling this why not use that context?in get service 

```
		utgen.NewUnitTestGenerator
```
here we can use the pre-defined context.


thus we won't need this signal handling which is a duplicate code

Reply:
Such as in like this? we have this in record
```
errGrp, _ := errgroup.WithContext(ctx)
```